pkg/resource/training_job/testdata/test_suite.yaml

```diff
@@ -1,17 +1,193 @@
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/describe/success_describe_debugger_variation.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/created_debugger_variation.yaml"
+      - name: "ReadOne=Stopping"
+        description: "Testing readOne right after delete, the status should be in Stopping"
+        given:
+          desired_state: "v1alpha1/readone/desired/stopping.yaml"
+          svc_api:
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/describe/stopping_describe.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/stopping.yaml"
+      - name: "ReadOne=StoppingDebugger"
+        description: "Testing readOne right after delete, the status should be in Stopping"
+        given:
+          desired_state: "v1alpha1/readone/desired/stopping_debugger_variation.yaml"
+          svc_api:
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/describe/stopping_describe_debugger_variation.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/stopping_debugger_variation.yaml"
+  - name: "Training job update tests"
+    description: "Testing the Update operation"
+    scenarios:
+      - name: "Update=NotSupported"
+        description: "This test checks if the controller throws error for update"
+        given:
+          desired_state: "v1alpha1/update/desired/updated_base.yaml"
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/error_on_update.yaml"
+          error: "not implemented"
+  - name: "Training job delete tests"
+    description: "Testing the delete operation"
+    scenarios:
+      - name: "Delete=Fail"
+        description: "This test checks if the condition is updated if delete fails and returns error"
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create_inprogress.yaml"
+          svc_api:
+            - operation: StopTrainingJobWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: Delete
+        expect:
+          latest_state: "v1alpha1/delete/observed/error_on_delete.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+      - name: "Delete=Successful"
+        description: "This test checks if the Training job is deleted successfully"
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create_inprogress.yaml"
+          svc_api:
+            - operation: StopTrainingJobWithContext
+            - operation: DescribeTrainingJobWithContext
+              error:
+                code: ValidationException
+                message: "Requested resource not found"
```

Comment:
Describe would still find the resource as there is no delete function in TrainingJob  
Our controller should work without the error response